controllers/reconcilers/kserve_networkpolicy_reconciler.go

```diff
@@ -98,47 +113,128 @@ func (r *KserveNetworkPolicyReconciler) createDesiredResource(isvc *kservev1beta
 			PolicyTypes: []v1.PolicyType{v1.PolicyTypeIngress},
 		},
 	}
-	return desiredNetworkPolicy, nil
+	return desiredNetworkPolicy
+}
+
+// allowOpenshiftIngressPolicy creates policy that grants ingress access through Openshift Routes to the services which
+// are part of a namespace under Service Mesh, but are not under Service Mesh control.
+func (r *KserveNetworkPolicyReconciler) allowOpenshiftIngressPolicy(isvc *kservev1beta1.InferenceService) *v1.NetworkPolicy {
+	return &v1.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NetworkPolicy",
+			APIVersion: "networking.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      openshiftIngressNetworkPolicyName,
+			Namespace: isvc.Namespace,
+			Labels:    map[string]string{"opendatahub.io/related-to": "RHOAIENG-1003"},
+		},
+		Spec: v1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			Ingress: []v1.NetworkPolicyIngressRule{
+				{
+					From: []v1.NetworkPolicyPeer{
+						{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"network.openshift.io/policy-group": "ingress"},
+							},
+						},
+					},
+				},
+			},
+			PolicyTypes: []v1.PolicyType{v1.PolicyTypeIngress},
+		},
+	}
+}
+
+// allowTrafficFromApplicationNamespaces creates combined network policy applied to pods in InferenceService namespace.
+// This set of policies allow traffic from:
+// - application namespace, where OpenDataHub and component service are deployed
+// - namespaces created by OpenDataHub where its component live
+// - traffic from other DataScienceProjects (namespaces created through dashboard)
+func (r *KserveNetworkPolicyReconciler) allowTrafficFromApplicationNamespaces(isvc *kservev1beta1.InferenceService) *v1.NetworkPolicy {
+	return &v1.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NetworkPolicy",
+			APIVersion: "networking.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      opendatahubNamespacesNetworkPolicyName,
+			Namespace: isvc.Namespace,
+			Labels:    map[string]string{"opendatahub.io/related-to": "RHOAIENG-1003"},
+		},
+		Spec: v1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			Ingress: []v1.NetworkPolicyIngressRule{
+				{
+					From: []v1.NetworkPolicyPeer{
+						{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"opendatahub.io/dashboard": "true"},
+							},
+						},
+						{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"kubernetes.io/metadata.name": r.appNamespace},
+							},
+						},
+						{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{"opendatahub.io/generated-namespace": "true"},
```

Comment:
These are labels added by odh-operator to every ns it creates.

Reply:
Can the user create the namespace himself somehow? Or use an existing one to deploy ISVC to?

Reply:
@israel-hdez `opendatahub.io/dashboard` is only applied on the DS projects.
if we want traffic going a bit wider by operator, application, and notebook's, better to have the later 2 added.
normally, the 3rd one should have the 2nd one covered, but in case, the label does not exsit (....)

Reply:
@ReToCode I guess user can do anything, but I am not sure this is supported scenario. Those policies take into account only the scenarios defined in godoc for this func:

```
// - application namespace, where OpenDataHub and component services are deployed
// - namespaces created by OpenDataHub where components live
// - traffic from other DataScienceProjects (namespaces created through dashboard)
```

I am not sure if the scenario where user creates a namespace on their own is something we should support / cover here. @danielezonca could you confirm?

Reply:
I guess there is no isolation between different notebooks also I guess a project admin could just misuse this right?

Reply:
> I am not sure if the scenario where user creates a namespace on their own is something we should support / cover here. @danielezonca could you confirm?

I think it will be possible in the future but the focus now is only "DS project" (aka namespaces with proper annotation)